src/server/scripts/Outland/zone_shadowmoon_valley.cpp

```diff
@@ -1429,7 +1453,74 @@ class npc_enraged_spirit : public CreatureScript
 
         void Reset() override { }
 
-        void EnterCombat(Unit* /*who*/) override { }
+        void EnterCombat(Unit* /*who*/) override
+        {
+            switch (me->GetEntry())
+            {
+                case NPC_ENRAGED_WATER_SPIRIT:
+                    _events.ScheduleEvent(EVENT_ENRAGED_WATER_SPIRIT, Seconds(0), Seconds(1));
+                    break;
+                case NPC_ENRAGED_FIRE_SPIRIT:
+                    _events.ScheduleEvent(EVENT_ENRAGED_FIRE_SPIRIT, Seconds(0), Seconds(1));
+                    break;
+                case NPC_ENRAGED_EARTH_SPIRIT:
+                    _events.ScheduleEvent(EVENT_ENRAGED_EARTH_SPIRIT, Seconds(0), Seconds(1));
+                    break;
+                case NPC_ENRAGED_AIR_SPIRIT:
+                    _events.ScheduleEvent(EVENT_ENRAGED_AIR_SPIRIT_CHAIN_LIGHTNING, Seconds(0), Seconds(1));
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+        _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_ENRAGED_WATER_SPIRIT:
+                        if (UpdateVictim())
+                            DoCastVictim(SPELL_STORMBOLT);
+                        _events.Repeat(Seconds(6), Seconds(9));
+                        break;
+                    case EVENT_ENRAGED_FIRE_SPIRIT:
+                        if (UpdateVictim())
+                            DoCastVictim(SPELL_FEL_FIREBALL);
+                        _events.Repeat(Seconds(7), Seconds(9));
+                        break;
+                    case EVENT_ENRAGED_EARTH_SPIRIT:
+                        if (UpdateVictim())
+                            DoCastVictim(SPELL_FIERY_BOULDER);
+                        _events.Repeat(Seconds(2), Seconds(3));
+                        break;
+                    case EVENT_ENRAGED_AIR_SPIRIT_CHAIN_LIGHTNING:
+                        if (UpdateVictim())
+                            DoCastVictim(SPELL_CHAIN_LIGHTNING);
+                        _events.ScheduleEvent(EVENT_ENRAGED_AIR_SPIRIT_HURRICANE, Seconds(3), Seconds(5));
+                        break;
+                    case EVENT_ENRAGED_AIR_SPIRIT_HURRICANE:
+                        if (UpdateVictim())
+                            DoCastVictim(SPELL_HURRICANE);
+                        _events.ScheduleEvent(EVENT_ENRAGED_AIR_SPIRIT_CHAIN_LIGHTNING, Seconds(10), Seconds(13));
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+		if (!UpdateVictim())
```

Comment:
it should be before of: https://github.com/TrinityCore/TrinityCore/pull/20225/files#diff-ba4f7087a55302e25b5ce01e7dcfdbbfR1479